cabal-install/src/Distribution/Client/CmdRun.hs

```diff
@@ -139,6 +145,13 @@ runAction flags@NixStyleFlags {..} targetAndArgs globalFlags
                ++ "You may wish to use 'build --only-dependencies' and then "
                ++ "use 'run'."
 
+            fullArgs <- getFullArgs
+            when (occursOnlyOrBefore fullArgs "+RTS" "--") $
+              warn verbosity $
+                  "Your RTS options are applied to cabal, not the executable. "
+               ++ "Use 'cabal run -- +RTS -N' to pass the RTS options "
```

Comment:
I'm not sure if hardcoding `-N` in the warning is a good idea.

Perhaps something like this is better:

> Your RTS options are applied to cabal, not the executable.
> Use '--' to separate cabal options from your executable options.
> For example, use 'cabal run -- +RTS -N' to pass the '-N' RTS option to your executable.

I also personally think it would be good to have newlines instead of one long line, but perhaps the warning will take too much vertical space then.

Reply:
I'd think the text will be wrapped by `warn` anyway. But someone needs to double-check.

Reply:
Yes, `warn` wraps text already.